src/url.cpp

```diff
@@ -122,16 +122,37 @@ namespace Url
             if (index != std::string::npos)
             {
                 query_ = path_.substr(index + 1);
+                remove_repeats(query_, '?');
+                remove_repeats(query_, '&');
                 path_.resize(index);
             }
 
             index = path_.find(';');
             if (index != std::string::npos)
             {
                 params_ = path_.substr(index + 1);
+                remove_repeats(params_, ';');
                 path_.resize(index);
             }
         }
     }
 
+    void Url::remove_repeats(std::string& str, const char chr)
+    {
+        size_t dest = 0;
+        bool seen = true;
+        for (size_t src = 0; src < str.length(); ++src)
+        {
+            if (!seen || (str[src] != chr))
+            {
+                str[dest] = str[src];
+                ++dest;
+            }
+            seen = str[src] == chr;
+        }
+        // Remove the last character if it happens to be chr
+        size_t length = ((dest > 0) && (str[dest - 1] == chr)) ? dest - 1 : dest;
+        str.resize(std::max(static_cast<size_t>(0), length));
```

Comment:
Instead of doing the cast in this case, I often will just explicitly specify the template parameter like so:
```c++
str.resize(std::max<size_t>(0, length));
```